src/otx/v2/adapters/torch/mmengine/mmaction/engine.py

```diff
@@ -0,0 +1,293 @@
+                PyTorch module, a dictionary containing the model configuration, or a string representing the path to
+                the model checkpoint file. Defaults to None.
+            img (Optional[Union[str, np.ndarray, list]], optional): The input image(s) to run inference on. Can be a
+                string representing the path to the image file, a NumPy array containing the image data, or a list of
+                NumPy arrays containing multiple images. Defaults to None.
+            checkpoint (Optional[Union[str, Path]], optional): The path to the checkpoint file to use for inference.
+                Defaults to None.
+            pipeline (Optional[Union[Dict, List]], optional): The data pipeline to use for inference. Can be a
+                dictionary containing the pipeline configuration, or a list of dictionaries containing multiple
+                pipeline configurations. Defaults to None.
+            device (Union[str, torch.device, None], optional): The device to use for inference. Can be a string
+                representing the device name (e.g. 'cpu' or 'cuda'), a PyTorch device object, or None to use the
+                default device. Defaults to None.
+            task (Optional[str], optional): The type of task to perform. Defaults to None.
+            batch_size (int, optional): The batch size to use for inference. Defaults to 1.
+            **kwargs: Additional keyword arguments to pass to the inference function.
+
+        Returns:
+            List[Dict]: A list of dictionaries containing the inference results.
+        """
+        from mmaction.apis.inferencers import MMAction2Inferencer
+
+        # Model config need data_pipeline of test_dataloader
+        # Update pipelines
+        if pipeline is None:
+            from otx.v2.adapters.torch.mmengine.mmaction.dataset import get_default_pipeline
+            pipeline = get_default_pipeline(subset="predict")
+        config = Config({})
+        if isinstance(model, torch.nn.Module) and hasattr(model, "_config"):
+            config = model._config  # noqa: SLF001
+        elif isinstance(model, dict) and "_config" in model:
+            config = model["_config"]
+        config["test_dataloader"] = {"dataset": {"pipeline": pipeline}}
+        if isinstance(model, dict):
+            model.setdefault("_config", config)
+        elif isinstance(model, torch.nn.Module):
+            model._config = config  # noqa: SLF001
+
+        if isinstance(checkpoint, Path):
+            checkpoint = str(checkpoint)
+        if task is not None and task != "VideoRecognition":
+            raise NotImplementedError
+
+        inferencer = MMAction2Inferencer(
+            rec=config,
+            rec_weights=checkpoint,
+            device=device,
+            input_format="rawframes",
+        )
+
+        return inferencer(img, batch_size=batch_size, **kwargs)
```

Comment:
So, currently we don't use the `model` even if it's `torch.nn.Module` which is already created and initialized?

Reply:
Yes, we decided to not use the MMXInferencerAPI. So, it'll be changed after phase1.